### PR TITLE
Cooldown Indicators

### PR DESCRIPTION
**Pull request by nisegami**

### CODE REVIEW

File: Assets/Game/Singletons/SpellMap.cs [hunk 1]
```diff
@@ -39,16 +47,45 @@ void Start()
...
 
-    private bool SpellReady(int playerNumber, SpellType spellType) { return cooldownTimers[playerNumber][(uint)spellType] <= 0.0f; }
+    private bool SpellReady(int playerNumber, SpellType spellType) { return cooldownTimers[playerNumber][(int)spellType] <= 0.0f; }
+
     private void RestartCooldown(int playerNumber, SpellType spellType)
+    {
+        SetCooldown(GetInitialCooldown(playerNumber, spellType), playerNumber, spellType);
+    }
+
+    private void DecrementCooldown(float deltaTime, int playerNumber, SpellType spellType)
```
Comment (BrandonShockley) [inline Assets/Game/Singletons/SpellMap.cs]:
```suggestion
    private void DecreaseCooldown(float deltaTime, int playerNumber, SpellType spellType)
```
I would argue that decrement suggests that you are decreasing the cooldown by a uniform amount each time.

File: Assets/Game/Singletons/SpellMap.cs [hunk 2]
```diff
@@ -19,6 +20,13 @@ private class Cooldowns
     [SerializeField]
     private Cooldowns[] cooldowns = new Cooldowns[(int)CharacterType._NUM_TYPES];
 
+
+    [SerializeField]
+    private Slider[] player1SpellIcons = null;
```
Comment (BrandonShockley) [inline Assets/Game/Singletons/SpellMap.cs]:
```suggestion
    private Slider[] player1CooldownSliders = null;
```

File: Assets/Game/Singletons/SpellMap.cs [hunk 3]
```diff
@@ -19,6 +20,13 @@ private class Cooldowns
...
+
+    [SerializeField]
+    private Slider[] player2SpellIcons = null;
```
Comment (BrandonShockley) [inline Assets/Game/Singletons/SpellMap.cs]:
```suggestion
    private Slider[] player2CooldownSliders = null;
```

**Review by BrandonShockley — CHANGES_REQUESTED** [review]
Also add the mask components to the icons. This will unfortunately require reworking the icon layout, as the mask component masks ALL children. I discovered a vertical layout component that you should be able to use to restack the icons. Leme know if this doesn't make sense.

**Comment by nisegami** [comment]
suggestions applied

**Comment by BrandonShockley** [comment]
I should probably stop making direct code suggestions. Check SpellMap.cs. You didn't change the places playerXCooldownSliders and DecreaseCooldown are used, so it doesn't compile. Also you'll need to assign the cooldown slider references in the SpellMap prefab in the main scene again since you changed the name in the script.

**Review by BrandonShockley — APPROVED** [review]

**Pull request merged**